docs/user-guide/manipulation/insert.md

```diff
@@ -232,3 +232,109 @@ runInsert $ insert (playlist chinookDb) $
   insertValues valuesToInsert
 putStrLn "See! I told you!"
 ```
+
+## `ON CONFLICT`
+
+Several backends (such as Postgres and SQLite) support `ON CONFLICT` subexpressions that specify
+what action to take when an `INSERT` statement conlicts with already present data.
+
+Beam support backend-agnostic `ON CONFLICT` statements via the `BeamHasInsertOnConflict`syntax. This
+class contains a new function to generate an `SqlInsert`. The `insertOnConflict` function can be
+used to attach `ON CONFLICT` actions to a `SqlInsert`.
+
+```haskell
+  insertOnConflict
+    :: Beamable table
+    => DatabaseEntity be db (TableEntity table)
+    -> SqlInsertValues be (table (QExpr be s))
+    -> SqlConflictTarget be table
+    -> SqlConflictAction be table
+    -> SqlInsert be table
+```
+
+The `SqlConflictTarget` specifies on which kinds of conflicts the action should run. You have a few options
+
+* `anyConflict` - run the action on any conflict
+* `conflictingFields` - run the action only when certain fields conflict
+* `conflictingFieldsWhere` - run the action only when certain fields conflict and a particular
+  expression evaluates to true.
+
+The `SqlConflictAction` specifies what to do when a conflict happens.
+
+* `onConflictDoNothing` - this cancels the insertion
+* `onConflictUpdateSet` - sets fields to new values based on the current values
+* `onConflictUpdateSetWhere` - sets fields to new values if a particular condition holds
+
+
+### Acting on any conflict
+
+A common use case of `ON CONFLICT` is to *upsert* rows into a database. *Upsertion* refers to only
+inserting a row if another conflicting row does not already exist. For example, if you have a new
```

Comment:
I think upsert refers to "Updating *or* inserting*". In this case you're inserting without updating.

Reply:
I guess that's just being super pedantic and probably different people don't agree on the specifics.

Reply:
Yeah i'll change that..